### DISCUSSION TITLE
Making an internal or private fork of a public repo

### DISCUSSION THREAD

**Discussion by hansmbakker** [community/community#34584] [Repositories]
Currently, if a repo is public, forks of it will also be public.

My organization would like to make private or internal forks of public repos.
- One usecase for this is e.g. forking public GitHub Actions to have control over security patches, while making it easy to keep up-to-date with upstream. The organization does not want to make it publicly visible what Actions they consume.
- Another usecase for this is to maintain private modifications to a public repo without losing track of upstream.

Without this functionality, the only way to achieve this is by [duplicating a repo](https://docs.github.com/en/repositories/creating-and-managing-repositories/duplicating-a-repository) which breaks the link between the repos. It makes staying up-to-date with upstream commits more cumbersome as explained in [this question on stackoverflow](https://stackoverflow.com/questions/60606902/how-to-make-a-fork-of-a-public-repo-private-and-compare-to-original-repo-on-gith).

Usecases from the [question on stackoverflow](https://stackoverflow.com/questions/60606902/how-to-make-a-fork-of-a-public-repo-private-and-compare-to-original-repo-on-gith):
> The primary use case is where you want to work on a fork and keep it updated with the source, but won't be making contributions back.
> _E.g. you are bespoking it for another project and the commits are too specific._

I know in the mentioned stackoverflow question there is an answer explaining a technical reason, but the point of this feedback item is to indicate the wish for functionality that currently does not exist. It is up to GitHub to see how they implement that.

**Comment by jfullstackdev** [comment]
maybe this will help 

https://stackoverflow.com/questions/10065526/github-how-to-make-a-fork-of-public-repository-private

https://gist.github.com/0xjac/85097472043b697ab57ba1b1c7530274

**Reply by hansmbakker** [reply]
Thank you for your reply - I found those indeed. That stackoverflow question was linked from the stackoverflow question I referenced. 

However, they rely on manually keeping the repos in sync - the link between them is broken this way and you do not an easy insight on upstream changes. So this is not what I was looking for - that is why I created this feedback item.

**Reply by hwittenborn** [reply]
I've seen that approach too, but I also don't want to have to manually check the upstream repository for changes.

**Comment by frankcorneliusmartin** [comment]
You can also do this from the ""Security" tab on your repo.
Create an advisory and from there you can create a private fork:
![image](https://user-images.githubusercontent.com/15981243/198634548-cbe95220-1a3d-4101-9b29-e2ddfd24a3d0.png)

**Reply by hwittenborn** [reply]
That's the workaround I'm using right now, but it's definitely far from ideal.
